src/entities/CommonUser.java

```diff
@@ -0,0 +1,76 @@
+package entities;
+
+import java.util.HashMap;
+
+public class CommonUser implements User {
+
+    final private String username;
+    final private String password;
+    private final HashMap<String, Double> portfolio;
+    private final HashMap<String, TransactionHistory> history;
+    private Double balance;
+
+    public void updatePortfolio(String ticker, Double amount) {
+        if (amount == 0) {
+            portfolio.remove(ticker);
+            return;
+        }
+        portfolio.put(ticker, amount);
+    }
```

Comment:
I think that the logic should belong to the use cases. According to the clean architecture this layer should only be comprised of classes with getter and setter functions.

Reply:
I see the potential mix-up. I have changed the logic of CommonUser and User accordingly. The use case interactors must also be changed